demo/test_demos.py

```diff
@@ -73,3 +73,43 @@ def test_demo(file, scalar_type):
             )
             == 0
         )
+
+
+@pytest.mark.parametrize("scalar_type", ["float64", "float32"])
+def test_demo_nvrtc(scalar_type):
+    """Test generated CUDA code with NVRTC."""
+    file = "Components"
+    opts = f"--scalar_type {scalar_type} --cuda"
+    if not sys.platform.startswith("linux"):
+        pytest.skip(reason="NVRTC support only tested on Linux")
+    else:
+        from nvidia import cuda_nvrtc
```

Comment:
We need an import skip on this test if cuda_nvrtc doesn't exist

Remove platform restrictions - I see no reason to artificially restrict this to Linux.

Reply:
So currently the cuda_nvrtc package won't install on MacOS, which is the rationale for restriction to Linux. This is also why I stuck the package under optional dependencies, because those are only installed in the CI for Linux.